### PR TITLE
fixed repurchasing course for 2nd time gives already purchased error

### PR DESCRIPTION
**Pull request by ahmadshafique**

### CODE REVIEW

File: ecommerce/extensions/order/utils.py [hunk 1]
```diff
@@ -193,10 +193,7 @@ def user_already_placed_order(user, product, site):
 
         orders_lines = OrderLine.objects.filter(product=product, order__user=user)
         for order_line in orders_lines:
-            if not UserAlreadyPlacedOrder.is_order_line_refunded(order_line):
-                if not order_line.product.is_course_entitlement_product:
-                    return True
-
+            try:
                 entitlement_uuid = order_line.attributes.get(option=entitlement_option).value
                 try:
```
Comment (hasnain-saeed) [inline ecommerce/extensions/order/utils.py:198]: Are we sure that order line attributes are same for refunded and non refunded order_lines. This piece of code was enclosed in `if not UserAlreadyPlacedOrder.is_order_line_refunded(order_line):`
This could also throw an error if it is not `is_course_entitlement_product`

File: ecommerce/extensions/order/utils.py [hunk 2]
```diff
@@ -193,10 +193,7 @@ def user_already_placed_order(user, product, site):
```
Comment (hasnain-saeed) [inline ecommerce/extensions/order/utils.py:193]: Can we test what `DISABLE_REPEAT_ORDER_CHECK_SWITCH_NAME` does?

**Review by hasnain-saeed — COMMENTED** [review]
We should avoid changing core functionality unless it becomes necessary

File: ecommerce/extensions/basket/utils.py [hunk 3]
```diff
@@ -159,10 +159,9 @@ def prepare_basket(request, products, voucher=None):
                 product.course_id
             )
             return basket
-
-        if product.is_enrollment_code_product or \
-                not UserAlreadyPlacedOrder.user_already_placed_order(user=request.user,
-                                                                     product=product, site=request.site):
+        if not UserAlreadyPlacedOrder.user_already_placed_order(user=request.user,
+                                                                 product=product, site=request.site) or \
+                product.is_enrollment_code_product:
```
Comment (tasawernawaz) [inline ecommerce/extensions/basket/utils.py:164]: why this change?

File: ecommerce/extensions/order/utils.py (hunk 2 above)
Comment (tasawernawaz) [reply ecommerce/extensions/order/utils.py:193]: yes, I believe this flag should handle our case.

**Review by tasawernawaz — CHANGES_REQUESTED** [review]

File: ecommerce/extensions/basket/utils.py (hunk 3 above)
Comment (ahmadshafique) [reply ecommerce/extensions/basket/utils.py:164]: was testing some other logic where product.is_enrollment_code_product was being False. There is 'OR' condition between them so I switched the sequence to first check if order was not already placed and had commented the other line. kindly ignore this change. i will revert. logic remains the same

File: ecommerce/extensions/order/utils.py (hunk 2 above)
Comment (ahmadshafique) [reply ecommerce/extensions/order/utils.py:193]: i does handle our case as checked but overrides logic to check for expired entitlements or if order is currently being refunded. it allows repeated order if they can be added to cart
